### ISSUE TITLE
What should be embedding size in ArcFace Loss

### ISSUE THREAD

**Issue by ashray21** [KevinMusgrave/pytorch-metric-learning#666]
I am using a pretrained pytorch model "convnext-large" and change its last layer to classify only 2 classes. What should be the value of the embedding_size in ArcFace loss function. Is it should be 2 or something else ?
Also if you can explain this.

**Comment by KevinMusgrave** [comment]
If you're replacing the last layer with `ArcFaceLoss`, then `num_classes=2` and `embedding_size` should be the size of the 2nd-last-layer's output. For example, if your model looks like this:

```
torch.nn.Sequential(
    torch.nn.Linear(512, 256),
    torch.nnLinear(256, 128),
    torch.nn.Linear(128, 2)
)
```

Then your new model should be:
```
torch.nn.Sequential(
    torch.nn.Linear(512, 256),
    torch.nnLinear(256, 128),
    ArcFaceLoss(num_classes=2, embedding_size=128)
)
```

**Comment by ashray21** [comment]
@KevinMusgrave 
I changed my model as you said and calculated ArcFace loss: `criterion = losses.ArcFaceLoss(2, embedding_size=128).` 
Now you said to get the class probablilites during inference I'm doing:
`for i, data in enumerate(training_dataloader):
            inputs,labels = data
            inputs = inputs.to(device)
            labels = labels.to(device)
            optimizer.zero_grad()  
            outputs = model(inputs)
            loss = criterion(outputs,labels)    
            loss.backward()
            optimizer.step()
            running_loss += loss.item()
            
            if loss_name == "arcface":
                normalized_features = F.normalize(outputs, dim=1)
                class_logits = torch.matmul(normalized_features, criterion.W)
                outputs = F.softmax(class_logits, dim=1)
           
            predicted = torch.argmax(outputs.data, 1)
`

Is this the correct way ?

**Comment by KevinMusgrave** [comment]
You can use `get_logits`:

```python
class_logits = criterion(outputs)
predicted = torch.argmax(class_logits)
```

Also during training, make sure to call `loss_optimizer.zero_grad()`

**Comment by ashray21** [comment]
@KevinMusgrave Thanks
